### PR TITLE
Dfl 9401 article links block

### PR DESCRIPTION
**Pull request by kevinhowbrook**
# Description

This PR adds the new Article Teaser block for selecting Article pages.

At the moment the blocks render out like so:

image

When compared with the designs, the circular images and text are missing, I'm not sure what these relate to on the article page model but I have been told it should be authors:

image

Link to sample test page: http://localhost:8000/en/campaigns/initial-test-bannered-campaign-with-fixed-title/ > Add a new article teaser block to the body

Related PRs/issues: #9401 

# Checklist

**Tests**
- ~~[ ] Is the code I'm adding covered by tests?~~

**Changes in Models:**
- [ ] ~~Did I update or add new fake data?~~
- [ ] ~~Did I squash my migration?~~
- [ ] ~~Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~

**Documentation:**
- [ ] ~~Is my code documented?~~
- [ ] ~~Did I update the READMEs or wagtail documentation?~~

### CODE REVIEW

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/article_teaser_block.html [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            <div class="blog-authors d-flex flex-wrap align-items-center">
+              <div class="tw-mx-2">
+                {% for author in card.article.authors.all %}
+                  {% with author_image_classes="-tw-ml-[0.7rem] tw-overflow-hidden tw-rounded-full tw-border-2 tw-border-white tw-inline-block tw-w-[40px] tw-h-[40px] tw-border-r d-flex align-items-center justify-content-center last:tw-mr-2" no_image_classes="tw-bg-[url('../_images/blog-author-placeholder.jpg')] tw-bg-cover "%}
+                    <div class="author-image {{ author_image_classes }} {% if author.author.image is None %} {{no_image_classes}} {% endif %}" aria-label="{{ author.author.name }}">
+                      {% if author.author.image %}
+                        <img src="{% image_url author.author.image "fill-100x100" %}" alt="{{ author.author.name }}" title="{{ author.author.name }}" />
+                      {% endif %}
+                    </div>
+                  {% endwith %}
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/article_teaser_block.html:30]: Argh... too bad [the fragment](https://github.com/mozilla/foundation.mozilla.org/blob/69568d67a91f08bb180eb3adfdf4fff38942d599/network-api/networkapi/templates/fragments/profile_images_stack.html) is not on `main` yet 🤦‍♂️  Not your fault of course.

**Review by tbrlpld — APPROVED** [review]
Thanks @kevinhowbrook, this is looking good to me. Just one question... and a comment that you can totally ignore.

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/article_teaser_block.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    article = blocks.PageChooserBlock(
+        required=False,
+        help_text='Page that this should link out to.',
+        target_model='wagtailpages.ArticlePage'
+    )
+
+
+class ArticleTeaserBlock(blocks.StructBlock):
+    cards = blocks.ListBlock(
+        ArticleTeaserCard(),
```
[the remark this answers is not in the document]
Comment (kevinhowbrook) [reply network-api/networkapi/wagtailpages/pagemodels/customblocks/article_teaser_block.py:14]: Yea exactly that - just trying to follow the pattern of flexibility on the project so far.

**Comment by tbrlpld** [comment]
Ah shit... I merged this prematurely. Sorry @kevinhowbrook Hope this does not break anything 😓

**Pull request merged**